### ISSUE TITLE
Fails to render a book containing a file with the `self-contained` option.

### ISSUE THREAD

**Issue by eitsupi** [quarto-dev/quarto-cli#360]
I tried to create a quarto book because I had accumulated several reports that I had set to be `self-contained: true` and distributed as a single HTML file.

In this case, the rendering failed due to the file containing the self-contained option.
I wonder if it is possible to make it so that invalid options are automatically ignored when rendering the entire project.
(However, such a behavior may also confuse the user, so it may be more appropriate to raise an error and prompt the user to remove the option before rendering.)

**Comment by jjallaire** [comment]
That's correct, you can't embed an entire website into a self-contained HTML file but rather only a single page (and HTML books are fundamentally websites). 

But perhaps you already realize that and the real bug here is not ignoring `self-contained` and/or providing a better error message. We will try to make these improvements soon!

**Comment by ylaizet** [comment]
Hi, It would be nice indeed to be able to render multiple qmd files in a single merge output HTML file. Either through a specific option of book rendering, or by allowing to specify a list of qmd files to render in a paramter of the yaml header. Very nice software by the way!

**Comment by eitsupi** [comment]
What I wanted to suggest in this issue is that it would be great if option conflicts could be resolved automatically, since there are different options to use when we want to distribute the individual files that make up a book individually, and when we want to build the entire project.

@ylaizet As mentioned above, my request is different from your request to create a Book in a single html format, so why not create a new issue about it?

**Comment by ylaizet** [comment]
> @ylaizet As mentioned above, my request is different from your request to create a Book in a single html format, so why not create a new issue about it?

You are totally right, I misread your point. I thought you where asking to allow self-contained option to work on book rendering. Sorry.

**Comment by mcanouil** [comment]
- https://github.com/quarto-dev/quarto-cli/issues/11316
